src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -223,5 +239,33 @@ public override Symbol AssociatedSymbol
                 return null;
             }
         }
+
+        internal override bool IsVirtualTupleField
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override sealed int GetHashCode()
+        {
+            return Hash.Combine(base.GetHashCode(), _name.GetHashCode());
+        }
+
+        public override sealed bool Equals(object obj)
+        {
+            return Equals(obj as TupleVirtualElementFieldSymbol);
+        }
+
+        public bool Equals(TupleVirtualElementFieldSymbol other)
```

Comment:
Right. I will just implement general Equal that would compare {containeingTuple, ID, Name}